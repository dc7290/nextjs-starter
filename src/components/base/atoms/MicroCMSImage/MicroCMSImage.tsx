import NextImage, { ImageProps, ImageLoader } from 'next/image'

type Props = Omit<ImageProps, 'src' | 'width' | 'height' | 'alt' | 'layout'> & {
  src: string
  width: number
  height: number
  alt: string
}

const defaultLoader: ImageLoader = ({ src, width, quality }) =>
  `${src}?auto=${quality !== undefined ? '' : 'compress,'}format&w=${width}${
    quality !== undefined ? `&q=${quality}` : ''
  }`

const MicroCMSImage = ({ loader = defaultLoader, className, ...imageProps }: Props) => {
  return (
    <div className={className}>
      <NextImage
        {...imageProps}
        loader={loader}
        layout="responsive"
        placeholder="blur"
        blurDataURL={`${imageProps.src}?auto=compress&w=10`}
      />
    </div>
  )
}

export default MicroCMSImage
