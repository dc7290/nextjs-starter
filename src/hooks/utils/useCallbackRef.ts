import { useCallback, useState } from 'react'

const useCallbackRef = <T extends HTMLElement>() => {
  const [current, setCurrent] = useState<T | null>(null)
  const ref = useCallback((element: T | null) => {
    if (element !== null) {
      setCurrent(element)
    }
  }, [])

  return { current, ref }
}

export default useCallbackRef
