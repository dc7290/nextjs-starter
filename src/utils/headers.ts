import { MicroCMSReqHeaders } from '../types/microCMS/headers'

const API_KEY = process.env.API_KEY

export const headers: MicroCMSReqHeaders = {
  'X-API-KEY': API_KEY ?? '',
}
