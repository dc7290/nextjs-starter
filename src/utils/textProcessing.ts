export const newLine = (text: string) => text.replace(/\n/g, '<br />')
export const singleLine = (text: string) => text.replace(/\n/g, '')

export const createAnchor = (text: string) =>
  text.replace(/\[([^\]]+)\]\(([^)]+)\)/g, '<a href="$2" target="_blank" rel="noreferrer">$1</a>')
