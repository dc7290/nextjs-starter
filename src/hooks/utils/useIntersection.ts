import { useEffect, useState } from 'react'

import useCallbackRef from './useCallbackRef'

const useIntersection = <T extends HTMLElement>() => {
  const [isIntersecting, setIsIntersecting] = useState(false)
  const { current, ref } = useCallbackRef<T>()
  useEffect(() => {
    const observer = new IntersectionObserver((entries) => {
      entries.forEach((entry) => {
        if (entry.isIntersecting) {
          setIsIntersecting(true)
        }
      })
    })
    if (current !== null) {
      observer.observe(current)
    }

    return () => {
      if (current !== null) {
        observer.unobserve(current)
      }
    }
  })
  return {
    isIntersecting,
    ref,
  }
}

export default useIntersection
