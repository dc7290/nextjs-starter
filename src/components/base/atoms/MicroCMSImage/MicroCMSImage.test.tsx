import React from 'react'

import { render } from '~/src/__tests__/utils'

import MicroCMSImage from './MicroCMSImage'

test('MicroCMSImage', () => {
  const { container } = render(<MicroCMSImage src="https://sample.png" width={640} height={360} alt="サンプル" />)
  expect(container).toMatchSnapshot()
})
