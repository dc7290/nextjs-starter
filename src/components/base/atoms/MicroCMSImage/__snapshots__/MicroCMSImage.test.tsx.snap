// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MicroCMSImage 1`] = `
<div>
  <div>
    <div
      style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
    >
      <div
        style="display: block; box-sizing: border-box; padding-top: 56.25%;"
      />
      <noscript />
      <img
        alt="サンプル"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(https://sample.png?auto=compress&w=10);"
      />
    </div>
  </div>
</div>
`;
